docs/content/developers/design-keuzes.md

```diff
@@ -391,3 +391,66 @@ Dit kan door het toevoegen van de laatste status (`STATUSTYPE` met het hoogste
 Er is echter behoefte aan een consistente manier om zaken af te sluiten. In
 deze oplossing worden zowel `ZAAK.einddatum` als `STATUSTYPE` gebruikt waarbij
 er geen onduidelijkheid meer ontstaat over hoe een `ZAAK` wordt afgesloten.
+
+## Mechanisme voor autorisatie
+
+De DSO schrijft voor om OAUTH2 te gebruiken op API niveau voor authenticatie
+en autorisatie. Kort samengevat is OAUTH2 te complex voor onze behoeften en
```

Comment:
Waarom is OAUTH2 te complex voor onze behoeften? Wat zijn onze behoeften? Waarom wordt het mogelijk gebruik van OAUTH2 hier al uitgesloten? OAUTH is een gestandaardiseerd protocol om een token (zoals ook een JWT token) te verkrijgen. Dus OAUTH2 en JWT sluiten elkaar niet uit maar kunnen elkaar aanvullen. Ik zou het gebruik van OAUTH2 optioneel maken.

Reply:
Ik had even gewacht tot de fieldlab van gisteren over het koppelpunt API authenticatie. Daar hebben we wel gekeken naar OAUTH2 profielen, en het uitgangspunt daar is dat OAUTH2 toegepast moet worden wanneer de eindgebruiker gegevens verstrekt aan derden. De use-case was het verstrekken van een inkomensindicatie uit UVW/Belastingdienst aan een woningcorporatie.

Onze situatie is daar heel verschillend in - de gemeente is eigenaar op de gegevens en heeft per definitie (gekaderd in doelbinding en AVG en al dat soort lol) permissie om de gegevens op te halen/uit te lezen om de zaken te kunnen behandelen. Er is geen burger/eindgebruiker buiten de organisatie die toegang kan verlenen tot gegevens - en dit is de hele premisse van OAUTH2. Externe partijen die zaken behandelen zijn door de gemeente gemandateerd om dit te doen en vallen technisch gezien binnen de organisatie.

Het grote probleem in onze architectuur van OAUTH2 is dat je een centrale autorisatieserver moet neerzetten, die gebruikt wordt door alle 380 gemeenten. Die centrale autorisatieserver moet dan vervolgens van alle authenticaties van alle gemeentes op de hoogte zijn. Dit brengt een enorme beheerslast met zich mee, introduceert een single-point of failure in een gedistribueerde architectuur terwijl het niets oplevert wat niet via JWT uitgegeven kan worden binnen de organisatie zelf.

Ze sluiten elkaar niet uit - de conclusie gisteren was dan ook dat er mogelijks een nood komt om die autorisaties van een eindgebruiker te delegeren, en het later invoeren van OAUTH als die nood er echt is, is ook nog technisch mogelijk.

Deze overwegingen zijn ook geformuleerd in `authenticatie-autorisatie.md`, in zekere vorm. Samengevat: OAUTH2 introduceert extra complexiteit zonder dat we er extra voordeel bij hebben.